src/dbnode/persist/fs/retriever.go

```diff
@@ -434,18 +435,17 @@ func (r *blockRetriever) fetchBatch(
 	reqs := r.filterAndCompleteWideReqs(allReqs, seeker, seekerResources,
 		retrieverResources)
 
-	var limitErr error
-	if err := r.queryLimits.AnyExceeded(); err != nil {
-		for _, req := range reqs {
+	for _, req := range reqs {
+		if err := r.queryLimits.AnyExceeded(); err != nil {
 			req.err = err
+			continue
 		}
-		return
-	}
 
-	for _, req := range reqs {
-		if limitErr != nil {
-			req.err = limitErr
+		select {
+		case <-req.stdCtx.Done():
+			req.err = req.stdCtx.Err()
```

Comment:
Can we do a nil check on the error? If Err() for whatever returns nil here, it'd be good to return an ErrCancelled or something since otherwise you'll read `nil` bytes rather than get an error to return to the client when you go to `convert.ToSegment(...)`.